src/lib/storage/storage_manager.cpp

```diff
@@ -390,141 +537,204 @@ void StorageManager::write_chunk_to_disk(const std::shared_ptr<Chunk>& chunk,
   }
 
   const auto segment_count = chunk->column_count();
-  for (auto segment_index = size_t{0}; segment_index < segment_count; ++segment_index) {
-    const auto abstract_segment = chunk->get_segment(static_cast<ColumnID>(static_cast<uint16_t>(segment_index)));
-    const auto dict_segment = dynamic_pointer_cast<DictionarySegment<int>>(abstract_segment);
-
-    write_dict_segment_to_disk(dict_segment, file_name);
+  for (auto segment_index = ColumnID{0}; segment_index < segment_count; ++segment_index) {
+    const auto abstract_segment = chunk->get_segment(segment_index);
+    _write_segment_to_disk(abstract_segment, file_name);
   }
 }
 
-void StorageManager::persist_chunks_to_disk(const std::vector<std::shared_ptr<Chunk>>& chunks,
-                                            const std::string& file_name) {
-  /*
-    TODO(everyone): Think about a proper implementation of a locking method. Each written file needs to be
-    locked prior to writing (and released afterwards). It was decided to use the mutex class by cpp.
-    (see https://en.cppreference.com/w/cpp/thread/mutex).
-  */
-  // file_lock.acquire();
-
-  auto chunk_segment_offset_ends =
-      std::vector<std::vector<uint32_t>>(StorageManager::_chunk_count, std::vector<uint32_t>());
-  auto chunk_offset_ends = std::array<uint32_t, StorageManager::_chunk_count>();
-  auto chunk_ids = std::array<uint32_t, StorageManager::_chunk_count>();
-
-  auto offset = uint32_t{_file_header_bytes};
-  for (auto chunk_index = uint32_t{0}; chunk_index < chunks.size(); ++chunk_index) {
-    const auto segment_offset_ends = generate_segment_offset_ends(chunks[chunk_index]);
-    offset += segment_offset_ends.back();
-
-    chunk_segment_offset_ends[chunk_index] = segment_offset_ends;
-    chunk_offset_ends[chunk_index] = offset;
-  }
-  // Fill all offset fields, that are not used with 0s.
-  for (auto chunk_index = chunks.size(); chunk_index < StorageManager::_chunk_count; ++chunk_index) {
-    chunk_offset_ends[chunk_index] = uint32_t{0};
-  }
+std::pair<uint32_t, uint32_t> StorageManager::_persist_chunk_to_file(const std::shared_ptr<Chunk> chunk,
+                                                                     ChunkID chunk_id,
+                                                                     const std::string& file_name) const {
```

Comment:
There is quite a bit of code duplication between persisting to an existing file and for a new file here. 
Either resolve code duplication or extract methods?